### ISSUE TITLE
Needs more testing

### ISSUE THREAD

**Issue by atjn** [putoutjs/minify#11]
I am using the [`minify`](https://www.npmjs.com/package/minify) package and I am trying to update from version 9, which uses terser, to version 11 which uses this package for minification.

After testing my code properly, I have found many issues with this package:
- #8
- will sometimes move statements into different logical blocks, significantly changing what the program does
- will sometimes downgrade strict equality checks to loose equality, which can change what the program does
- will hoist variables even though they have side-effects, which changes what the program does
- turns async functions into sync functions, which can remove intentional parallelization
- ..I find new bugs for every new file I test it with

I could write out full bug reports for everything I found, but because there are so many bugs, I don't think it makes sense to try and manually find every bug and manually fix them, I think you need to write a much more comprehensive test/fuzzing system to make sure you catch all these issues. Until then, I don't feel comfortable using this minifier, as I never know when some new code will be minified incorrectly.

My program is also measurably slower. I think it might be because this package often rewrites `let`/`const` to `var`. While `var` is a little shorter, I believe it is much harder for the JS parser to isolate, and therefore the program becomes slower. I don't think a minifier should change the variable declaration type.

It also does some weird stuff, like changing `" "+v` to `" ${v}"`. The new code is 2 characters longer than the original.

In summary, I cannot use this package because my programs break if I do, and even if you fix the specific issues that I have identified right here, I am not confident that the minifier works correctly for the next code I write. I do not trust it unless it is tested much more thoroughly.

I am sorry to open such a negative issue, I am not trying to be annoying, I just wanted to let you know about the isues I found. I hope you find this helpful.  Feel free to close this issue at any time :)

**Comment by coderaiser** [comment]
There is a list of rules that used both in 🐊**Putout** and **Minify**, it made for a reason to make both tools better, so if you have ideas for issue - create it, and we will talk about it, anything can be configured and improved.
